### PR TITLE
nixos/proxychains: add package option

### PR DESCRIPTION
**Pull request by chuangzhu**
###### Description of changes

Addresses https://github.com/NixOS/nixpkgs/issues/136093#issuecomment-1479987887

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/programs/proxychains.nix [hunk 1]
```diff
@@ -51,6 +51,16 @@ in {
 
       enable = mkEnableOption (lib.mdDoc "installing proxychains configuration");
 
+      package = mkOption {
```
Comment (jian-lin) [inline nixos/modules/programs/proxychains.nix:54]: FYI, mkPackageOption

**Review by jian-lin — APPROVED** [review]

**Comment by Freed-Wu** [comment]
I try `programs.proxychains.package = pkgs.proxychains-ng;`.
`pkgs.proxychains-ng` has a zsh completion script `_proxychains4`.
However,`programs.proxychains.package = pkgs.proxychains-ng;` only add `/run/current-system/sw/bin/proxychains4`, will not add `/run/current-system/sw/share/zsh/site-functions/_proxychains4`. Is it a bug?

**Comment by chuangzhu** [comment]
It does have `share/zsh/site_functions/_proxychains4` though... Seems to be an upstream typo: https://github.com/rofl0r/proxychains-ng/blob/0c795085fa36d3cc5ba2d30cb5de77e07407f3b6/Makefile#L15

**Comment by Freed-Wu** [comment]
Yes, it is.

**Comment by Freed-Wu** [comment]
How about add a symlink `ln -s proxychains4 proxychains`? Archlinux do this thing, <https://archlinux.org/packages/extra/x86_64/proxychains-ng/> Why not nixos do it, too?

**Pull request merged**